packages/table/src/table-column.js

```diff
@@ -294,6 +294,11 @@ export default {
       }
     });
 
+    // Deprecation warning for renderHeader property
+    if (this.renderHeader) {
+      console.warn('[Element Warn][Table Column]\'render-header\' property will be deprecated in next major version. please use scoped-slot \'header\' instead.');
```

Comment:
I find `render-header` is widely used. Maybe we can't deprecate this attribute in next major version. We should give users enough time to upgrade. 
A potential warning info is like this:
> Comparing to `render-header`, scoped-slot header is easier to use. We recommend users to use scoped-slot header.